pkg/gator/reader/read_resources.go

```diff
@@ -146,6 +155,63 @@ func ToTemplate(scheme *runtime.Scheme, u *unstructured.Unstructured) (*template
 	return template, nil
 }
 
+// ToSyncSet converts an unstructured SyncSet into a SyncSet struct.
+func ToSyncSet(scheme *runtime.Scheme, u *unstructured.Unstructured) (*syncsetv1alpha1.SyncSet, error) {
+	if u.GroupVersionKind().Group != syncsetv1alpha1.GroupVersion.Group || u.GroupVersionKind().Kind != "SyncSet" {
+		return nil, fmt.Errorf("%w", gator.ErrNotASyncSet)
+	}
+
+	s, err := ToStructured(scheme, u)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", gator.ErrAddingSyncSet, err)
+	}
+
+	syncSet, isSyncSet := s.(*syncsetv1alpha1.SyncSet)
+	if !isSyncSet {
+		return nil, fmt.Errorf("%w: %T", gator.ErrAddingSyncSet, syncSet)
+	}
+
+	return syncSet, nil
+}
+
+// ToConfig converts an unstructured Config into a Config struct.
+func ToConfig(scheme *runtime.Scheme, u *unstructured.Unstructured) (*configv1alpha1.Config, error) {
+	if u.GroupVersionKind().Group != configv1alpha1.GroupVersion.Group || u.GroupVersionKind().Kind != "Config" {
+		return nil, fmt.Errorf("%w", gator.ErrNotAConfig)
+	}
+
+	s, err := ToStructured(scheme, u)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", gator.ErrAddingConfig, err)
+	}
+
+	config, isConfig := s.(*configv1alpha1.Config)
+	if !isConfig {
+		return nil, fmt.Errorf("%w: %T", gator.ErrAddingConfig, config)
+	}
+
+	return config, nil
+}
+
+// ToGVKManifest converts an unstructured GVKManifest into a GVKManifest struct.
+func ToGVKManifest(scheme *runtime.Scheme, u *unstructured.Unstructured) (*gvkmanifestv1alpha1.GVKManifest, error) {
```

Comment:
These To___ and Is___ functions start to feel repetitive but i'm not sure how deep in the weeds of reflection it's worth getting to streamline them